model/src/main/scala/explore/model/enum/AppTab.scala

```diff
@@ -4,20 +4,42 @@
 package explore.model.enum
 
 import cats.data.NonEmptyList
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.PosInt
 import lucuma.core.util.Enumerated
 
-sealed abstract class AppTab(val title: String) extends Product with Serializable
+sealed abstract class AppTab(
+  val title:       String,
+  val buttonGroup: ButtonGroup,
+  val groupOrder:  PosInt
```

Comment:
If I understand correctly, `groupOrder` is being used to define the order of tabs within a group. Since order is preserved in groups after `groupBy`, we might not need this.

Reply:
Is it so that `groupOrder` is ordering inside the group? While the ordering of groups is defined by
the `ButtonGroup` enumerated?

Reply:
Correct, groupOrder provides ordering within groups, while the ordering of groups is provided by the Enumerated `Order` instance.

@rpiaggio Yes, ordering is preserved, so technically we don't need this. But, the direction of comments on the PR seemed to be towards explicitly stating groups and orders so I included it.

Personally, this commit seems like overkill to me. The button group/order will seldom be changed and, when it is, it should be verified by the developer. It seems reasonable to require the AppTabs be `Order`ed in the order they will be displayed, and use a simple method of grouping them. I think that even including the `ButtonGroup` Enumerated is extra machinery that does little to provide safety. The developer can still assign a button to the wrong `ButtonGroup` and the compiler can't catch that. But, that is just one opinion. 😄

Reply:
@rpiaggio Carlos and I were discussing this, and think going back to the "int in AppTab" is probably a good balance between simplicity and control. If we do this, I'll add a comment for extra clarification. What do you think?

Reply:
Ok, sounds good for this use case.

Reply:
Yeah it seems it is getting to complex for a small gain in safety